pkg/kyc/component.go

```diff
@@ -304,6 +307,163 @@ func (c *component) ValidateUserInSocialRealm(ctx context.Context, userID string
+	accessToken, err := c.tokenProvider.ProvideToken(ctx)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Can't get OIDC token", "err", err.Error())
+		return err
+	}
+
+	var confRealm = ctx.Value(cs.CtContextRealm).(string)
+	return c.validateUserBasicID(ctx, accessToken, confRealm, c.socialRealmName, userID, user)
+}
+
+func (c *component) validateUserBasicID(ctx context.Context, accessToken string, confRealm string, targetRealm string, userID string, user apikyc.UserRepresentation) error {
+	var operatorName = ctx.Value(cs.CtContextUsername).(string)
+
+	// Get the user from Keycloak
+	kcUser, err := c.keycloakClient.GetUser(accessToken, targetRealm, userID)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "CreateAccreditations: can't get Keycloak user", "err", err.Error(), "realm", targetRealm, "user", userID)
+		return err
+	}
+
+	//Creation of the new accreditation HARDCODED
+	newAccreds := map[string]string{}
+	typeAccred := "basic-check"
+	validityAccred := "150y"
+	expiry, err := c.convertDurationToDate(ctx, validityAccred)
+	if err != nil {
+		return err
+	}
+	creationMillis := time.Now().UnixNano() / int64(time.Millisecond)
+	var newAccreditationJSON, _ = json.Marshal(keycloakb.AccreditationRepresentation{
+		Type:           &typeAccred,
+		ExpiryDate:     expiry,
+		CreationMillis: &creationMillis,
+	})
+	newAccreds[typeAccred] = string(newAccreditationJSON)
+
+	var mergedAccreds = evaluateCurrentAccreditations(kcUser.GetAttribute(constants.AttrbAccreditations), newAccreds)
+
+	// Merge all attributes
+	for _, accValue := range newAccreds {
+		mergedAccreds = append(mergedAccreds, accValue)
+	}
+	kcUser.SetAttribute(constants.AttrbAccreditations, mergedAccreds)
+
+	keycloakb.ConvertLegacyAttribute(&kcUser)
+
+	// Some parameters might not be updated by operator
+	user.ID = &userID
+	user.Email = nil
+	user.PhoneNumber = nil
+	user.EmailVerified = nil
+	user.PhoneNumberVerified = nil
+	user.Username = kcUser.Username
+
+	// Gets user from database
+	dbUser, err := c.usersDBModule.GetUserDetails(ctx, targetRealm, userID)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Failed to get user from database", "err", err.Error())
+		return err
+	}
+
+	user.ExportToDBUser(&dbUser)
+	user.ExportToKeycloak(&kcUser)
+
+	var now = time.Now()
+	err = c.keycloakClient.UpdateUser(accessToken, targetRealm, userID, kcUser)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Failed to update user through Keycloak API", "err", err.Error())
+		return err
+	}
+
+	// Store user in database
+	err = c.usersDBModule.StoreOrUpdateUserDetails(ctx, targetRealm, dbUser)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Can't store user details in database", "err", err.Error())
+		return err
+	}
+
+	// Store check in database
+	var validation = dto.DBCheck{
+		Operator:  &operatorName,
+		DateTime:  &now,
+		Status:    ptr("VERIFIED"),
+		Type:      ptr("IDENTITY_CHECK"),
+		Nature:    ptr("BASIC_CHECK"),
```

Comment:
BASIC_CHECK / basic-check ?

Reply:
Currrently for the physical check, we set in the "Nature" field : "PHYSICAL_CHECK", even if the accreditation type is "physical-check". So I did exactly the same. Maybe @harture has an opinion about this naming.